src/components/icon/js/iconService.js

```diff
@@ -422,8 +422,25 @@
     */
    function cacheIcon( id ) {
 
-     return function updateCache( icon ) {
-       iconCache[id] = isIcon(icon) ? icon : new Icon(icon, config[id]);
+     return function updateCache( _icon ) {
+       var icon = isIcon(_icon) ? _icon : new Icon(_icon, config[id]);
+
+       //clear id attributes to prevent aria issues
+       var elem = icon.element;
+       elem.removeAttribute('id');
+
+       var children = elem.childNodes;
+
+       for (var i=0; i<children.length;i++) {
+         var child = children[i];
+         if(child.nodeName==='path') {
```

Comment:
@DevVersion  I found a similar test case https://jsperf.com/document-vs-element-queryselectorall-performance . queryselectorall looks very performant when it is narrowed on a certain element. In this test case it did 215,335 operations in ie per second